tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/DbScopeFactory.cs

```diff
@@ -224,7 +224,21 @@ private static string GetServiceName(Tracer tracer, string dbTypeName)
 
             private static DbCommandCache.TagsCacheItem GetTagsFromConnectionString(IDbCommand command)
             {
-                var connectionString = command.Connection?.ConnectionString;
+                IDbConnection dbConnection = null;
```

Comment:
Wondering out loud if there's a cheaper way for us to identify that this comes from a `DbDataSource`? 🤔 I think we need this code either way, but it seems expensive to be throwing an exception (twice) for every `DbCommand`...

Reply:
Maybe it's better to have the `GetType().FullName` thing for known types that throws. WDYT?

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3730/commits/8e6eeac5bb8fa58a613de9c6cb21b098ca8039ec